### DISCUSSION TITLE
how to customize web-app template and server templates.go file

### DISCUSSION THREAD

**Discussion by Lynne42** [dexidp/dex#2150] [Q&A]
I want to customize web-app template , I have found a way in [this issue](https://github.com/dexidp/dex/issues/1525)
but when I customize the template based on the design draft, I found that I need to modify the template.go server, what should I do

dex version: v2.22.0

**Comment by sagikazarmark** [comment]
Here is the documentation: https://dexidp.io/docs/templates/

Watch out for changes in the next version though: https://github.com/dexidp/website/pull/94

2.22 is rather old, I'd suggest upgrading to the latest version.

**Reply by Lynne42** [reply]
Thank you very much, I have customized the template according to this document, and it works well after testing

But I found that the template.go file needs to be modified to add some functions to the template. I don’t know how to configure the modified template file in config-dev.yaml

**Reply by Lynne42** [reply]
![image](https://user-images.githubusercontent.com/46595240/120263092-9c390980-c2cd-11eb-8fb4-de02b213832a.png)
For example, I want to modify the lower to upper here and convert the string to uppercase. I want to pass the strings.ToUpper function through template.go

![image](https://user-images.githubusercontent.com/46595240/120263400-30a36c00-c2ce-11eb-88d0-f978d90656a6.png)

**Reply by sagikazarmark** [reply]
Sadly, that's not configurable right now, you need a custom build for that. I've opened #2151 to track that issue.

**Reply by Lynne42** [reply]
In fact, I want to be able to use a custom template.go, just like I can use a custom web by configuring frontend

I don’t know if it’s what this [gRPC API document](https://dexidp.io/docs/api/) says, because I don’t know the go language, I haven’t practiced it according to this document, and even I don’t fully understand dex, just make a UI page based on it.

**Reply by sagikazarmark** [reply]
Go is a compiled language, so you can't replace a single source code file on the fly. You need to recompile.
